provider/source/data/debug/currency/patterns@1/ar.json

```diff
@@ -1,5 +1,6 @@
 {
-  "unit_patterns": {
-    "Other": "{0} {1}"
+  "patterns": {
+    "specials": [],
+    "other": "\u0002\u0005 "
```

Comment:
As you can see, the space shifted to the end of the pattern instead in the middle.

For me the PR looks good, let's merge it after the LongCurrencyFormatter PR just to ensure that the test result from this PR are equivalent.

Reply:
This is the string encoding of `icu_pattern`. The escaped values are indices, and the space is the content.